src/uct/cuda/cuda_copy/cuda_copy_iface.c

```diff
@@ -455,15 +450,101 @@ static uct_iface_internal_ops_t uct_cuda_copy_iface_internal_ops = {
     .ep_is_connected       = uct_base_ep_is_connected
 };
 
+static ucs_status_t
+uct_cuda_copy_init_per_ctx_rscs(const uct_cuda_copy_iface_t *iface,
+                                CUcontext cuda_ctx,
+                                unsigned long long ctx_id,
+                                uct_cuda_copy_per_ctx_rsc_t *ctx_rsc)
+{
+    ucs_status_t status;
+    ucs_mpool_params_t mp_params;
+    ucs_memory_type_t src, dst;
+
+    ucs_mpool_params_reset(&mp_params);
+    mp_params.elem_size       = sizeof(uct_cuda_copy_event_desc_t);
+    mp_params.elems_per_chunk = 128;
+    mp_params.priv_size       = sizeof(CUcontext);
+    mp_params.max_elems       = iface->config.max_cuda_events;
+    mp_params.ops             = &uct_cuda_copy_event_desc_mpool_ops;
+    mp_params.name            = "CUDA EVENT objects";
+    status = ucs_mpool_init(&mp_params, &ctx_rsc->cuda_event_desc);
+    if (UCS_OK != status) {
+        ucs_error("mpool creation failed");
+        return UCS_ERR_IO_ERROR;
+    }
+
+    ucs_memory_type_for_each(src) {
+        ucs_memory_type_for_each(dst) {
+            ctx_rsc->queue_desc[src][dst].stream = 0;
+            ucs_queue_head_init(&ctx_rsc->queue_desc[src][dst].event_queue);
+        }
+    }
+
+    ctx_rsc->short_stream = 0;
+    ctx_rsc->cuda_ctx     = cuda_ctx;
+    ctx_rsc->ctx_id       = ctx_id;
+
+    return UCS_OK;
+}
+
+ucs_status_t uct_cuda_copy_get_per_ctx_rscs(uct_cuda_copy_iface_t *iface,
+                                            CUcontext cuda_ctx,
+                                            uct_cuda_copy_per_ctx_rsc_t **ctx_rsc_p)
```

Comment:
Please remove the declaration from the .h file, and make the function static.
```suggestion
static ucs_status_t
uct_cuda_copy_get_per_ctx_rscs(uct_cuda_copy_iface_t *iface, CUcontext cuda_ctx,
                               uct_cuda_copy_per_ctx_rsc_t **ctx_rsc_p)
```